unit-tests/py/rspy/libci.py

```diff
@@ -146,42 +146,56 @@ def __init__( self, source, line_prefix, context ):
         self.derive_tags_from_path( source )
 
     def derive_config_from_text( self, source, line_prefix ):
-        regex = r'^' + line_prefix + r'(\S+)((?:\s+\S+)*?)\s*(?:#\s*(.*))?$'
-        for context in file.grep( regex, source ):
-            match = context['match']
+        regex = r'^' + line_prefix + r'([^\s:]+)(:(\S+))?((?:\s+\S+)*?)\s*(?:#\s*(.*))?$'
+        for line in file.grep( regex, source ):
+            match = line['match']
             directive = match.group( 1 )
-            text_params = match.group( 2 ).strip()
+            directive_context = match.group( 3 )
+            text_params = match.group( 4 ).strip()
             params = [s for s in text_params.split()]
-            comment = match.group( 3 )
+            comment = match.group( 5 )
+            if directive_context:
+                if not directive_context.startswith('!') and directive_context != self.context:
+                    log.d( "directive", line['line'], "ignored because of context mismatch with running context",
+                           self.context)
```

Comment:
Let me know what you think of these messages and if they are necessary in your opinion

Reply:
Not necessary IMO, but good to have in case you want to debug -- maybe leave the line in, but commented?